### PR TITLE
Use an actual HTML to Markdown parser.

### LINKED ISSUE #3: Transform links into markdown
It would be super useful if HTML links would be transformed to MarkDown as well.

Would you merge a PR that adds this functionality @npcode? If so I'll create a PR for it.

### PR DESCRIPTION
**Pull request by roelvanduijnhoven**
This PR uses an actual Markdown parser to do the HTML conversion to Markdown. The upsides of that are lots of other constructs are now converted to markdown. These include headings, strong, italic, lists, blockquotes, images and some more.

The dependency to `to-markdown` should probably not be in repo; but could't find another way to do it. First time I do anything with a Chrome extension ;).

Curious to hear your thoughts. 

This would fix #3.

### CODE REVIEW

**Comment by roelvanduijnhoven** [comment]
Btw. A sample:

----

> ## Oorzaken van de schaarste[[bewerken](/w/index.php?title=Hongerwinter&action=edit&section=1 "Bewerk dit kopje: Oorzaken van de schaarste")]
> 
> De oorzaak van de voedselschaarste in het westen van Nederland werd achteraf als gevolg beschouwd van drie oorzaken[[bron?]](/wiki/Wikipedia:Bronvermelding#Bron_gevraagd "Wikipedia:Bronvermelding"):
> 
> *   Door de bevrijding van zuidelijk Nederland was er geen aanvoer van [steenkool](/wiki/Steenkool "Steenkool") uit de Limburgse mijnen meer mogelijk. Vanuit Duitsland kwam slechts zeer beperkt brandstof. Ook daar was behoefte aan brandstof voor het dagelijks leven en de oorlogsvoering.
> *   De regeringsopdracht voor de [Algemene Spoorwegstaking in 1944](/wiki/Spoorwegstaking_van_1944 "Spoorwegstaking van 1944"), afgekondigd op [17 september](/wiki/17_september "17 september") door Nederlandse [regering in Londen](/wiki/Londens_kabinet "Londens kabinet"), waardoor het transport in Nederland in elk geval voor enkele weken voor een groot deel werd stilgelegd.
> *   De pressiemiddelen van de Duitse bezetter om deze staking te doen breken, tezamen met het algemene Duitse beleid. Relatief eenvoudig, door blokkades op te werpen bij de [Afsluitdijk](/wiki/Afsluitdijk "Afsluitdijk"), de bruggen over de IJssel en de Friese/Overijsselse IJsselmeerhavens en het niet meer afgeven van transportvergunningen, werd voor een periode van zes weken het vervoer van voedsel naar het westen geblokkeerd. Het veroorzaakte in West-Nederland een hongerramp van catastrofale omvang.
>
> -- https://nl.wikipedia.org/wiki/Hongerwinter#Oorzaken_van_de_schaarste

File: manifest.json
Comment (eungjun-yi) [inline manifest.json:4]: I think you missed this file.

**Review by eungjun-yi — CHANGES_REQUESTED** [review]

**Comment by eungjun-yi** [comment]
Thanks for your awesome PR! I'll merge this after review.

File: manifest.json
Comment (roelvanduijnhoven) [reply manifest.json:4]: Doh! Stupid me. I updated the PR with the library.

**Comment by roelvanduijnhoven** [comment]
Updated the PR. Thanks for the review.

**Comment by eungjun-yi** [comment]
It seems that relative URI references doesn't work. I think we need to convert them to absolute form.

Here is an example:

> ## 36.1 Introduction
> 
> Since version 3.1, Spring Framework provides support for transparently adding caching into an existing Spring application. Similar to the [transaction](transaction.html "17. Transaction Management") support, the caching abstraction allows consistent use of various caching solutions with minimal impact on the code.
> 
> As from Spring 4.1, the cache abstraction has been significantly improved with the support of [JSR-107 annotations](cache.html#cache-jsr-107 "36.4 JCache (JSR-107) annotations") and more customization options.
>
> -- https://docs.spring.io/spring/docs/current/spring-framework-reference/html/cache.html

**Comment by eungjun-yi** [comment]
And I think the readability as a plain text could be improved if the links are represented in label form.

As-is:

```
This is [the well-known example site](http://example.org "Example Domain").
```

To-be:

```
This is [the well-known example site][1].

[1]: http://example.org "Example Domain"
```

The rendered results of both of above are exactly same:

This is [the well-known example site](http://example.org "Example Domain").

This is [the well-known example site][1].

[1]: http://example.org "Example Domain"

**Comment by roelvanduijnhoven** [comment]
Just back from vacation :mountain:. Thanks for the review.

* Good catch with the relative URLs. We should definitely fix that. 
* I don't see too much trouble with either representation for links. I myself simply paste the response; and do not look at its source. I think this should not block this PR. 

I'll do my best to update this PR to address the relative URLs. Should manage to do so within one week.

**Comment by roelvanduijnhoven** [comment]
Allright! Found some time to this.

The some contents you pasted:

> ## 36.1 Introduction
> 
> Since version 3.1, Spring Framework provides support for transparently adding caching into an existing Spring application. Similar to the [transaction](https://docs.spring.io/spring/docs/current/spring-framework-reference/html/transaction.html "17. Transaction Management") support, the caching abstraction allows consistent use of various caching solutions with minimal impact on the code.
> 
> As from Spring 4.1, the cache abstraction has been significantly improved with the support of [JSR-107 annotations](https://docs.spring.io/spring/docs/current/spring-framework-reference/html/cache.html#cache-jsr-107 "36.4 JCache (JSR-107) annotations") and more customization options.
>
> -- https://docs.spring.io/spring/docs/current/spring-framework-reference/html/cache.html

**Review by eungjun-yi — APPROVED** [review]
Looks good to me. I'll handle the markdown syntax issue for links myself.

Thank you for the great contribution! :+1:

**Comment by roelvanduijnhoven** [comment]
Thanks.

When will this land in a new release of the extension?

**Comment by eungjun-yi** [comment]
@roelvanduijnhoven Sorry for late response. I've just uploaded the new version in the chrome web store and you can see it soon!

**Pull request merged**
